magefiles/magefile.go

```diff
@@ -738,64 +744,51 @@ func AppendFrameworkDescribeGoFile(specFile string) error {
 
 }
 
-func newSprayProxy() (*SprayProxy, error) {
+func newSprayProxy() (*sprayproxy.SprayProxyConfig, error) {
 	sprayProxyUrl := os.Getenv("QE_SPRAYPROXY_HOST")
 	sprayProxyToken := os.Getenv("QE_SPRAYPROXY_TOKEN")
 	if sprayProxyUrl == "" || sprayProxyToken == "" {
 		return nil, fmt.Errorf("env var QE_SPRAYPROXY_HOST is not set")
 	}
-	return NewSprayProxy(sprayProxyUrl, sprayProxyToken), nil
+	return sprayproxy.NewSprayProxyConfig(sprayProxyUrl, sprayProxyToken)
 }
 
 func registerPacServer() error {
-	pacControllerRoute, err := GetPacRoute()
-	if err != nil {
-		return err
-	}
-	pacControllerHost := fmt.Sprintf("https://%s", pacControllerRoute.Spec.Host)
-
-	sprayproxy, err := newSprayProxy()
+	var err error
+	sprayProxyConfig, err = newSprayProxy()
 	if err != nil {
-		return fmt.Errorf("failed to register PaC server: %+v", err)
+		return fmt.Errorf("failed to set up SprayProxy credentials: %+v", err)
 	}
-	_, err = sprayproxy.RegisterServer(pacControllerHost)
-
+	_, err = sprayProxyConfig.RegisterServer()
 	if err != nil {
-		klog.Error("Failed to register pac server", pacControllerHost)
-		return err
+		return fmt.Errorf("error when registering PaC server %s to SprayProxy server %s: %+v", sprayProxyConfig.PaCHost, sprayProxyConfig.BaseURL, err)
 	}
 	// for debugging purposes
-	servers, err := sprayproxy.GetServers()
+	servers, err := sprayProxyConfig.GetServers()
```

Comment:
it will make more sense if it print out the current registered Pac servers instead all of Pac servers registered in Sprayproxy server.

Reply:
addressed in https://github.com/redhat-appstudio/e2e-tests/pull/832/commits/869bb8cc77fc9bb347029c2bf18e8158f7a8da4d